src/tests/Directory.Build.targets

```diff
@@ -570,4 +569,28 @@
           Condition="'$(_WillCLRTestProjectBuild)' == 'true' and '$(CLRTestKind)' == 'BuildAndRun'"
           DependsOnTargets="ComputeResolvedFilesToPublishList;LinkNative" />
 
+  <Target Name="LocateNativeCompiler"
```

Comment:
There seems to be a lot of duplication of this logic. Would it be possible to have this logic in a shared file under `eng` and import it in all places that need it?